File: Delphi/uMainForm.pas
unit uMainForm;

interface

uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
  Vcl.Controls, Vcl.Forms, Vcl.Dialogs, cxClasses, dxReport, cxGraphics,
  cxLookAndFeels, cxLookAndFeelPainters, Vcl.Menus,  cxButtons,
  dxBackend, cxControls, cxStyles, cxCustomData, cxFilter, cxData,
  cxDataStorage, cxEdit, cxNavigator, dxDateRanges, dxScrollbarAnnotations,
  Data.DB, cxDBData, cxGridLevel, cxGridCustomView, cxGridCustomTableView,
  cxGridTableView, cxGridDBTableView, cxGrid, FireDAC.Comp.DataSet,
  FireDAC.Comp.Client, dxBackend.ConnectionString.JSON.DataSet,
  dxBackend.ConnectionString.JSON, Vcl.StdCtrls, dxmdaset, uData, cxContainer,
  cxLabel, cxRadioGroup, cxGroupBox;

type
  TMainForm = class(TForm)
    dxReport1: TdxReport;
    dxBackendDataConnectionManager1: TdxBackendDataConnectionManager;
    btnDisplayDesigner: TcxButton;
    btnDisplayReport: TcxButton;
    dxBackendDataConnectionManager1dxBackendDataSetJSONConnection1: TdxBackendDataSetJSONConnection;
    itmProducts: TdxBackendDataSetCollectionItem;
    itmCategories: TdxBackendDataSetCollectionItem;
    rbtnGroupLocalization: TcxRadioGroup;
    rbtnSelectEnglishLocalization: TcxRadioButton;
    rbtnSelectGermanLocalization: TcxRadioButton;
    btnGroupDisplayDialog: TcxGroupBox;
    procedure FormCreate(Sender: TObject);
    procedure btnDisplayDesignerClick(Sender: TObject);
    procedure btnDisplayReportClick(Sender: TObject);
    procedure rbtnSelectEnglishLocalizationClick(Sender: TObject);
    procedure rbtnSelectGermanLocalizationClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  MainForm: TMainForm;

implementation

{$R *.dfm}

procedure TMainForm.FormCreate(Sender: TObject);
begin
  // Switch to German localization when the application starts
  dxReport1.Language := 'de-DE';
end;

procedure TMainForm.btnDisplayDesignerClick(Sender: TObject);
begin
  // Display the DevExpress Report Designer dialog
  dxReport1.ShowDesigner;
end;

procedure TMainForm.btnDisplayReportClick(Sender: TObject);
begin
  // Display the DevExpress Report Viewer dialog
  dxReport1.ShowViewer;
end;

procedure TMainForm.rbtnSelectEnglishLocalizationClick(Sender: TObject);
begin
  // Switch to English localization
  dxReport1.Language := 'en-US';
end;

procedure TMainForm.rbtnSelectGermanLocalizationClick(Sender: TObject);
begin
  // Switch to German localization
  dxReport1.Language := 'de-DE';
end;

end.

File: CPB/uMainForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "uMainForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxButtons"
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxDataStorage"
#pragma link "cxDBData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxGrid"
#pragma link "cxGridCustomTableView"
#pragma link "cxGridCustomView"
#pragma link "cxGridDBTableView"
#pragma link "cxGridLevel"
#pragma link "cxGridTableView"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "cxNavigator"
#pragma link "cxStyles"
#pragma link "dxDateRanges"
#pragma link "dxReport"
#pragma link "dxReport.Backend"
#pragma link "dxReport.ConnectionString.JSON"

#if defined(_WIN64)
  #pragma link "dxReport.ConnectionString.JSON.DB.O"
#else
  #pragma link "dxReport.ConnectionString.JSON.DB.OBJ"
#endif


#pragma link "dxScrollbarAnnotations"
#pragma link "dxmdaset"
#pragma link "cxContainer"
#pragma link "cxLabel"
#pragma resource "*.dfm"
TMainForm *MainForm;
//---------------------------------------------------------------------------
__fastcall TMainForm::TMainForm(TComponent* Owner)
	: TForm(Owner)
{

}
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
void __fastcall TMainForm::btnShowDesignerClick(TObject *Sender)
{
	dxReport1->ShowDesigner();
}
//---------------------------------------------------------------------------

void __fastcall TMainForm::btnViewReportClick(TObject *Sender)
{
	dxReport1->ShowViewer();
}
//---------------------------------------------------------------------------



void __fastcall TMainForm::btnSetLanguageUSClick(TObject *Sender)
{
	dxReport1->Language = "en-US";
}
//---------------------------------------------------------------------------

void __fastcall TMainForm::btnSetLanguageDEClick(TObject *Sender)
{
	dxReport1->Language = "de-DE";
}
//---------------------------------------------------------------------------

